### PR TITLE
302: Add the without availabilities filter on planning search

### LINKED ISSUE #302: Do not display users without availability in the planning
=> Ajouter une case "Afficher aussi les resources sans disponibilité" dans la recherche du planning
=> cocher la case par défaut quand on arrvie au planning par les boutons "disponibilités de MES bénévoles"
=> Si la case n'est pas cochée, ne pas récupérer ni afficher les assets & users sans dispo sur la période affichée

### PR DESCRIPTION
**Pull request by dju45**
Closes #302

### CODE REVIEW

File: src/Form/Type/PlanningSearchType.php [hunk 1]
```diff
@@ -114,6 +114,10 @@ public function buildForm(FormBuilderInterface $builder, array $options): void
                     'data-actions-box' => 'true',
                 ],
             ])
+            ->add('withoutAvailabilities', CheckboxType::class, [
```
Comment (mRoca) [inline src/Form/Type/PlanningSearchType.php:117]: Starting the parameter value with "display", "only", or "hide" would be easier to understand. What do you think about `displayWithoutAvailabilities` ?

File: src/Repository/CommissionableAssetRepository.php [hunk 2]
```diff
@@ -85,6 +85,10 @@ public function findByFilters(array $formData, bool $onlyIds = false): array
             $qb = $this->addAvailabilityBetween($qb, $formData['availableFrom'], $formData['availableTo'], CommissionableAssetAvailability::class, 'asset');
         }
 
+        if (($formData['withoutAvailabilities'] ?? false) === false && empty($formData['availableFrom']) && empty($formData['availableTo'])) {
+            $qb = $this->addAvailabilityBetween($qb, $formData['from'], $formData['to'], CommissionableAssetAvailability::class, 'asset');
```
Comment (mRoca) [inline src/Repository/CommissionableAssetRepository.php:89]: Caution: you cannot use this method here: what we want is `Do not display users without availability in the planning`, not `Display only ressources available ON EVERY slot between from and to`, you must create a new function in order to get assets and users `with AT LEAST one "not red" slot  between from and to`

File: src/Repository/CommissionableAssetRepository.php [hunk 3]
```diff
@@ -85,6 +85,10 @@ public function findByFilters(array $formData, bool $onlyIds = false): array
             $qb = $this->addAvailabilityBetween($qb, $formData['availableFrom'], $formData['availableTo'], CommissionableAssetAvailability::class, 'asset');
         }
 
+        if (($formData['withoutAvailabilities'] ?? false) === false && empty($formData['availableFrom']) && empty($formData['availableTo'])) {
```
Comment (vincentchalamon) [inline src/Repository/CommissionableAssetRepository.php:88]: Please use Yoda condition (cf. [Symfony Coding Standards](https://symfony.com/doc/current/contributing/code/standards.html#structure)):
```suggestion
        if (false === ($formData['withoutAvailabilities'] ?? false) && empty($formData['availableFrom']) && empty($formData['availableTo'])) {
```

File: src/Repository/UserRepository.php [hunk 4]
```diff
@@ -110,6 +110,10 @@ public function findByFilters(array $formData, bool $onlyIds = false): array
             $qb = $this->addAvailabilityBetween($qb, $formData['availableFrom'], $formData['availableTo'], UserAvailability::class, 'user', $availableStatuses);
         }
 
+        if (($formData['withoutAvailabilities'] ?? false) === false && empty($formData['availableFrom']) && empty($formData['availableTo'])) {
```
Comment (vincentchalamon) [inline src/Repository/UserRepository.php:113]: Same:
```suggestion
        if (false === ($formData['withoutAvailabilities'] ?? false) && empty($formData['availableFrom']) && empty($formData['availableTo'])) {
```

**Comment by mRoca** [comment]
Closed in favor of #367

**Pull request closed without merging**